repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilterChain.java

```diff
@@ -100,8 +107,35 @@ public void startFilterChain(ServletRequest servletRequest, ServletResponse serv
         servletRequest.setAttribute("filterChainAvailableForRequest", filterChainAvailable);
         servletRequest.setAttribute("http://openrepose.org/requestUrl", ((HttpServletRequest) servletRequest).getRequestURL().toString());
         servletRequest.setAttribute("http://openrepose.org/queryParams", servletRequest.getParameterMap());
+        MutableHttpServletRequest wrappedRequest = MutableHttpServletRequest.wrap((HttpServletRequest) servletRequest);
+        splitRequestHeaders(wrappedRequest);
 
-        doFilter(servletRequest, servletResponse);
+        doFilter(wrappedRequest, servletResponse);
+    }
+
+    private void splitRequestHeaders(MutableHttpServletRequest request) {
```

Comment:
Does it make more sense for this method (and associated methods) to be inside the MutableHttpServletRequest wrapper instead? Then we could still split the headers in this class, but anyone using our wrappers could also split headers at will. We still get the explicit splitHeaders() call in the PowerFilterChain, but the PowerFilterChain won't reuse this logic, and it's semi-verbose in al already crowded class.